### PR TITLE
Make embedded target host verification consider all known user projects

### PR DESCRIPTION
**Pull request by benasher44**
Even though the analysis step that actually finds the host targets, to which to copy the embedded targets' pod targets, searches all available user projects, the step that verifies that the embedded targets' host targets are in the Podfile did not. This meant that this verification could incorrectly throw an `Informative`, even if the Podfile was correct in cases where the embedded targets lived in a sub-project.

This + CocoaPods/Xcodeproj#396 should fix the host target for sub-project embedded target identification issue #5622, as well as #5685.

### CODE REVIEW

**Comment by benasher44** [comment]
I updated the Gemfile to point at my branch for CocoaPods/Xcodeproj#396, since this is required for this PR. I'll revert that change and just update the Gemfile.lock for Xcodeproj once CocoaPods/Xcodeproj#396 ships.

**Comment by benasher44** [comment]
Looks like travis is having trouble installing json for 1 check. No changes to that dep here.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
-* None.  
+* Improve host target detection for embedded targets
+  in sub-projects
```
Comment (segiddins) [inline CHANGELOG.md]: period & trailing two spaces

**Comment by segiddins** [comment]
👍

**Comment by selectersky** [comment]
👍

**Comment by benasher44** [comment]
Unsure what to do about the execution_output.txt failure.

**Comment by segiddins** [comment]
@benasher44 it's a spurious failure -- as far as I'm concerned, this build is passing

**Comment by benasher44** [comment]
Hmm. That rebase did not go as expected. Fixing…

**Comment by benasher44** [comment]
Okay that was odd, but the diff is sane again 😅

**Comment by benasher44** [comment]
Ack: looks like the same spurious failure again, although in the 3rd job you have to look at the raw output (unsure why).

**Comment by benasher44** [comment]
@segiddins this is ready to go on my end! 😃

**Pull request merged**